### PR TITLE
refine tests for some layers

### PR DESCRIPTION
**Pull request by wzhongyuan**
## What changes were proposed in this pull request?

Refine some tests with new approach

## How was this patch tested?

unit test

### CODE REVIEW

**Comment by wzhongyuan** [comment]
@yiheng  Refined the testes gradually to avoid conflicts with others

**Comment by hkvision** [comment]
Currently Keras side is still adding more layers and serialization tests. Will this influence you?

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/nn/PermuteSpec.scala [hunk 1]
```diff
@@ -41,3 +44,11 @@ class PermuteSpec extends KerasBaseSpec {
   }
 
 }
+class PermuteSerialTest extends ModuleSerializationTest {
```
Comment (hkvision) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/nn/PermuteSpec.scala]: Add a new line after the previous bracket?

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/nn/ReshapeSpec.scala [hunk 2]
```diff
@@ -58,3 +61,12 @@ class ReshapeSpec extends KerasBaseSpec {
...
 }
+
+class ReshapeSerialTest extends ModuleSerializationTest {
+  override def test(): Unit = {
+    val layer = Reshape[Float](Array(4, 15), inputShape = Shape(3, 4, 5))
+    layer.build(Shape(2, 3, 4, 5))
+    val input = Tensor[Float](2, 3, 4, 5).apply1(_ => Random.nextFloat())
+    runSerializationTest(layer, input)
+  }
+}
```
Comment (hkvision) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/keras/nn/ReshapeSpec.scala]: Need a new line at the end of the file. Or style test will fail?

**Comment by wzhongyuan** [comment]
@hkvision  it doesn't matter for now, as we still keep the old tests before we updated for all layers
BTW, new lines added

**Comment by wzhongyuan** [comment]
PR validation passed http://172.168.2.101:8080/job/BigDL-PR-Validation-v3/158/

**Pull request merged**